applicationLibrary/CMakeLists.txt

```
@@ -11,6 +11,12 @@ file(
 
 add_library(molecularDynamicsLibrary STATIC ${LIB_MD_SRC})
 
+if (MD_FLEXIBLE_MODE MATCHES  "SINGLESITE|singlesite")
+    target_compile_definitions(molecularDynamicsLibrary PUBLIC MD_FLEXIBLE_MODE=0 SINGLESITE=0 MULTISITE=1)
+elseif(MD_FLEXIBLE_MODE MATCHES "MULTISITE|multisite")
```

I'd say this is still not fool-prove enough ;)
Typically what you want is to lower-case the input and then compare against the expected string. So something like:
```suggestion
string( TOLOWER "${MD_FLEXIBLE_MODE}" MD_FLEXIBLE_MODE_lower )
if (MD_FLEXIBLE_MODE_lower MATCHES "singlesite")
    target_compile_definitions(molecularDynamicsLibrary PUBLIC MD_FLEXIBLE_MODE=0 SINGLESITE=0 MULTISITE=1)
elseif(MD_FLEXIBLE_MODE_lower MATCHES "multisite")
```

---

done in [53d6fad](https://github.com/AutoPas/AutoPas/pull/778/commits/53d6fad3ea476758572a183c8d9e849437fcdf9e)